src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machine> {
```
no need for MachineService to have its own type parameter, it implements interface with T as Machine